### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by Ichimonji10**
Add unit tests for all manner of code in NailGun. Improve coverage from 64% to
82%. There is still room for significantly improving the test suite, and these
added tests simply represent some of the low-hanging fruit.

Test results against a downstream system:

    $ nosetests tests/foreman/api/test_multiple_paths.py
    ...S...S......S...............................................................S...S..........................................................................................................................................SSS..S.S.SSSSSS.S..SS.SSSSSSSSSSSSSSSSSSSSSSSSSSSS
    ----------------------------------------------------------------------
    Ran 271 tests in 487.348s

    OK (SKIP=47)

Also, document the new `test-coverage` make target.

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -43,11 +43,17 @@ def test_no_value(self):
...
 
     def test_existing_value(self):
-        """Assert that an existing content-type is not overridden."""
+        """Assert that no content-type is provided if one is set."""
         kwargs = {'headers': {'content-type': ''}}
         client._set_content_type(kwargs)  # pylint:disable=protected-access
         self.assertEqual(kwargs, {'headers': {'content-type': ''}})
 
+    def test_files_in_kwargs(self):
+        """Assert that no content-type if provided if files are given."""
```
Comment (omaciel) [inline tests/test_client.py]: s/if/is

File: tests/test_entities.py [hunk 2]
```diff
@@ -232,50 +232,90 @@ class CreatePayloadTestCase(TestCase):
...
+                entities.UserGroup,
+                entities.User,
+            )
+        ]
+        entities_.extend([
+            (entities.SyncPlan, {'organization': 1}),
+            (entities.ContentViewPuppetModule, {'content_view': 1}),
+        ])
+        for entity, params in entities_:
+            if version_info < (3, 4):
```
Comment (omaciel) [inline tests/test_entities.py]: Could you add a comment explaining why this check is being performed? It isn't obvious to me from this diff :/

File: tests/test_entity_fields.py [hunk 3]
```diff
@@ -101,6 +107,41 @@ def test_mac_address_field(self):
             '^([0-9A-F]{2}[:]){5}[0-9A-F]{2}$'
         )
 
+    def test_dict_field(self):
+        """Test :meth:`nailgun.entity_fields.DictField.gen_value`.
+
+        Assert that an emtpy dict is returned by default. There are very few
```
Comment (omaciel) [inline tests/test_entity_fields.py]: s/emtpy/empty

File: tests/test_entity_fields.py [hunk 4]
```diff
@@ -101,6 +107,41 @@ def test_mac_address_field(self):
...
+        occurrences of dict fields in the entity classes, so it is hard to
+        intelligently produce a randomized value that will be of use in a wide
+        variety of entities. Isntead, those few entities override or extend
```
Comment (omaciel) [inline tests/test_entity_fields.py]: s/Isntead/Instead

**Comment by omaciel** [comment]
APT pending comments

**Comment by Ichimonji10** [comment]
Thank you for the comments, @omaciel. Much appreciated.

File: Makefile [hunk 5]
```diff
@@ -5,6 +5,7 @@ help:
 	@echo "  help           to show this message"
 	@echo "  lint           to run flake8 and pylint"
 	@echo "  test           to run unit tests"
+	@echo "  test-coverage  to run unit tests and measure test coverage"
```
Comment (elyezer) [inline Makefile]: Thank you, I've missed that from my PR.

**Comment by elyezer** [comment]
ACK

@omaciel's comments are addressed. Time to merge it.

**Pull request merged**